core/kernel/ubsan.c

```diff
@@ -111,7 +111,7 @@ void __ubsan_handle_vla_bound_not_positive(struct vla_bound_data *data,
 					   unsigned long bound);
 void __ubsan_handle_load_invalid_value(struct invalid_value_data *data,
 				       unsigned long val);
-void __ubsan_handle_nonnull_arg(struct nonnull_arg_data *data, size_t arg_no);
+void __ubsan_handle_nonnull_arg(struct nonnull_arg_data *data);
```

Comment:
You can't do this unconditionally because it breaks gcc < 6.2.1
```
$ arm-linux-gnueabihf-gcc --version | head -1
arm-linux-gnueabihf-gcc (Ubuntu/Linaro 5.4.0-6ubuntu1~16.04.4) 5.4.0 20160609
$ make -j9 -s CFG_CORE_SANITIZE_UNDEFINED=y CROSS_COMPILE32=arm-linux-gnueabihf-
core/kernel/ubsan.c:114:6: error: conflicting types for built-in function ‘__ubsan_handle_nonnull_arg’ [-Werror]
 void __ubsan_handle_nonnull_arg(struct nonnull_arg_data *data);
      ^
core/kernel/ubsan.c:229:6: error: conflicting types for built-in function ‘__ubsan_handle_nonnull_arg’ [-Werror]
 void __ubsan_handle_nonnull_arg(struct nonnull_arg_data *data)
      ^
cc1: all warnings being treated as errors
mk/compile.mk:146: recipe for target 'out/arm-plat-vexpress/core/kernel/ubsan.o' failed
make: *** [out/arm-plat-vexpress/core/kernel/ubsan.o] Error 1
make: *** Waiting for unfinished jobs....
$
```